src/utils/TicketRuleConfigurationFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace staabm\PHPStanTodoBy\utils;
+
+use PHPStan\DependencyInjection\Container;
+use staabm\PHPStanTodoBy\utils\github\GitHubTicketStatusFetcher;
+use staabm\PHPStanTodoBy\utils\jira\JiraTicketStatusFetcher;
+
+final class TicketRuleConfigurationFactory
+{
+    private Container $container;
+
+    public function __construct(Container $container)
+    {
+        $this->container = $container;
+    }
+
+    public function create(): TicketRuleConfiguration
+    {
+        /** @var array{ticket:array{resolvedStatuses: list<string>, keyPrefixes: list<string>, tracker: 'github'|'jira'}} $extensionParameters */
+        $extensionParameters = $this->container->getParameter('todo_by');
+
+        $parameters = $extensionParameters['ticket'];
+        $resolvedStatuses = $parameters['resolvedStatuses'];
+        $keyPrefixes = $parameters['keyPrefixes'];
+        $tracker = $parameters['tracker'];
+
+        if ('jira' === $tracker) {
+            $fetcher = $this->container->getByType(JiraTicketStatusFetcher::class);
+
+            return new TicketRuleConfiguration(
+                $fetcher::getKeyPattern(),
+                $resolvedStatuses,
+                $keyPrefixes,
+                $fetcher,
+            );
+        }
+
+        if ('github' === $tracker) {
+            $fetcher = $this->container->getByType(GitHubTicketStatusFetcher::class);
+
+            return new TicketRuleConfiguration(
+                $fetcher::getKeyPattern(),
+                ['closed'],
+                ['#'],
+                $fetcher,
+            );
+        }
+    }
```

Comment:
else: `throw new RuntimeException("Unsupported tracker type: $type");`

Reply:
It's currently redundant, thanks to this comment that declares possible `tracker` values:
```php
/** @var array{ticket:array{resolvedStatuses: list<string>, keyPrefixes: list<string>, tracker: 'github'|'jira'}} $extensionParameters */
```
Phpstan knows that the code terminates in one of these two conditionals above - in fact, it would show an error if I added this `throw`.

But I'm not sure if having this `@var` doc is a good idea. It might become a lie if we don't keep it in sync with parameters schema in `extension.neon`. But without it phpstan reports issues (on level 9 - which we currently use), unless I validate all parameters at run-time what makes factory cluttered with no significant benefits. 

Does anyone have an idea what's the best way to tackle this problem?

Reply:
I am fine with reducing to level 8. 

We should not use inline `@var`